components/collapse/collapse.js

```diff
@@ -0,0 +1,129 @@
+import React, {
+  Component,
+  Children,
+  cloneElement,
+  PropTypes,
+} from 'react';
+
+import { getClassNamesWithMods } from '../_helpers';
+import CollapseItem from './collapseItem';
+
+const getNormalizedActiveKey = ({ activeKey = [] }) => {
+  return activeKey instanceof Array ? activeKey : [activeKey];
+};
+
+class Collapse extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeKey: getNormalizedActiveKey(props),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ('activeKey' in nextProps) {
+      this.setState({
+        activeKey: getNormalizedActiveKey(nextProps),
+      });
+    }
+  }
+
+  handleItemClick = (e, key) => {
```

Comment:
if I will use `=>`, [babel](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-0&targets=&browsers=&builtIns=false&code=class%20C1%20%7B%0A%20%20f1()%20%7B%0A%20%20%20%20this.a%20%3D%201%3B%0A%20%20%7D%0A%20%20%0A%20%20f2%20%3D%20()%20%3D%3E%20%7B%0A%20%20%20%20this.a%20%3D%201%3B%0A%20%20%7D%0A%7D) move it to constructor, and create new function for each instance, this is the same if we do `bind` in constructor.